src/structs.rs

```diff
@@ -107,6 +107,63 @@ impl From<&AssetCode> for BaseField {
     }
 }
 
+/// Amount value type
+#[tagged_blob("AMOUNT_VALUE")]
+#[derive(
+    Debug,
+    Clone,
+    Copy,
+    PartialEq,
+    Default,
+    Hash,
+    Eq,
+    From,
+    Into,
+    Add,
+    AddAssign,
+    Sub,
+    SubAssign,
+    Mul,
+    MulAssign,
+    Div,
+    DivAssign,
+    Rem,
+    RemAssign,
+    PartialOrd,
+    Ord,
+    Sum,
+)]
+pub struct AmountValue(pub(crate) u128);
```

Comment:
Why `AmountValue` rather than just `Amount`? Or `RecordAmount`?

Reply:
Done in 8dd1c7b26dbb7c31f013d778033e396011d4f665